### DISCUSSION TITLE
Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.

### DISCUSSION THREAD

**Discussion by ham0nd** [pschatzmann/ESP32-A2DP#351] [Q&A] [answered]
hi,pschatmann.
@pschatmann
i am using the libraries of esp32-a2dp and arduino-audio-tools.
my board is esp32-wrover-e,and it have been enabled psram.
i want to use the esp32 to play the music from flash and bluetooth source.but when i start sending the stream from bluetooth after finish playing the music from flash, the esp32 will be crushed and show `Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.`
but Strangely, it is normal operation when i play bluetooth music alone or play flash music.
here is my code and the crush log.
thx ^_^
![image](https://user-images.githubusercontent.com/43801159/211140020-77f6581c-9b5d-4b59-aa25-7cd8e6ff9dcb.png)

`#include <Arduino.h>
#include "BluetoothA2DPSink.h"
#include "AudioTools.h"
#include "StarWars30.h"

uint8_t channels = 1;
uint16_t sample_rate = 22050;

I2SStream i2s; // Output to I2S
MemoryStream music(StarWars30_raw, StarWars30_raw_len);
StreamCopyT<int16_t> copier(i2s, music); // copies sound into i2s

BluetoothA2DPSink a2dp_sink;

#define DEVICE_NAME "test"
#define Bit_Clock_BCLK 27
#define Word_Select_WS 26
#define Serial_Data_SD 25

bool play_state = false;

void audio_data_callback(const uint8_t *data, size_t item_size)
{
  size_t i2s_bytes_written = 0;
  i2s_write(I2S_NUM_0, (void *)data, item_size, &i2s_bytes_written, portMAX_DELAY);
}

void setup()
{
  Serial.begin(115200);
  Serial.printf("Sample MP3 playback begins...\n");
  auto config = i2s.defaultConfig(TX_MODE);
  config.sample_rate = sample_rate;
  config.channels = channels;
  config.bits_per_sample = 16;
  i2s.begin(config);
  a2dp_sink.set_stream_reader(audio_data_callback, false);
  a2dp_sink.start((char *)DEVICE_NAME);
}

void loop()
{
  if (play_state == true)
  {
    if (!copier.copy())
    {
      i2s.end();
      stop();
      Serial.printf("MP3 done\n");
      play_state = false;
    }
  }

  if (Serial.available())
  {
    String command = Serial.readString();
    Serial.println("start playing");
    if (command == "play")
    {
      play_state = true;
    }
  }
}`

**Comment by ham0nd** [comment, marked as answer]
@pschatzmann

**Reply by pschatzmann** [reply]
I suggest that you analyse the stacktrace. This will give you the offending line...
ps. the psram is not used anywere in your sketch, so you can also switch it off. 
The stop() is just running an endless loop to stop any further processing, so you should really decide what you really want to do
